libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/arrow/api/MapboxRouteArrowView.kt

```diff
@@ -23,43 +30,128 @@ import com.mapbox.navigation.ui.maps.route.arrow.model.RouteArrowState
 class MapboxRouteArrowView(private val options: RouteArrowOptions) {
 
     /**
-     * Renders a UpdateRouteArrowVisibilityState applying view side effects based on the data
+     * Renders a [ArrowVisibilityChangeValue] applying view side effects based on the data
      * it contains.
      *
      * @param style a valid map style object
      * @param state a state containing data for applying the view side effects.
      */
-    fun render(style: Style, state: RouteArrowState.UpdateRouteArrowVisibilityState) {
+    fun render(style: Style, visibilityChange: ArrowVisibilityChangeValue) {
         initializeLayers(style, options)
 
-        state.getVisibilityChanges().forEach {
+        visibilityChange.layerVisibilityModifications.forEach {
             updateLayerVisibility(style, it.first, it.second)
         }
     }
 
     /**
-     * Renders a UpdateManeuverArrowState applying view side effects based on the data
-     * it contains.
+     * Renders a Expected<UpdateManeuverArrowValue, RouteArrowUpdateError> applying view side
+     * effects based on the data it contains.
      *
      * @param style a valid map style object
      * @param state a state containing data for applying the view side effects.
      */
-    fun render(style: Style, state: RouteArrowState.UpdateManeuverArrowState) {
+    fun renderManeuverUpdate(
+        style: Style,
+        state: Expected<UpdateManeuverArrowValue, RouteArrowUpdateError>
+    ) {
         initializeLayers(style, options)
 
-        state.getVisibilityChanges().forEach {
-            updateLayerVisibility(style, it.first, it.second)
+        when (state) {
+            is Expected.Failure -> {
+            }
```

Comment:
Is this empty on purpose? Should we at least log (or leave a quick comment in the code)?

Reply:
Log statement added.